tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/generators/cpp/s3control/S3ControlRestXmlCppClientGenerator.java

```diff
@@ -172,5 +173,35 @@ protected SdkFileEntry generateARNSourceFile(final ServiceModel serviceModel) th
 
         return makeFile(template, context, fileName, true);
     }
+
+    private void addShape(final ServiceModel serviceModel,
```

Comment:
is it possible to add this to the base class 'RestXmlCppClientGenerator' so that it can be reused for s3 and s3 control, since this method is identical in both

Reply:
I like the addShape method and it would be great to reuse it

Reply:
I've updated it, thank you!